lib/ts/recipe/passwordless/recipe.ts

```diff
@@ -108,6 +113,378 @@ export default class Recipe extends RecipeModule {
                     emailDelivery: undefined,
                     smsDelivery: undefined,
                 });
+
+                let otpOrLink: string[] = [];
+                let emailOrPhone: string[] = [];
+
+                if (Recipe.instance.config.flowType === "MAGIC_LINK") {
+                    otpOrLink.push("link");
+                } else if (Recipe.instance.config.flowType === "USER_INPUT_CODE") {
+                    otpOrLink.push("otp");
+                } else {
+                    otpOrLink.push("otp");
+                    otpOrLink.push("link");
+                }
+
+                if (Recipe.instance.config.contactMethod === "EMAIL") {
+                    emailOrPhone.push("email");
+                } else if (Recipe.instance.config.contactMethod === "PHONE") {
+                    emailOrPhone.push("phone");
+                } else {
+                    emailOrPhone.push("email");
+                    emailOrPhone.push("phone");
+                }
+
+                const allFactors: string[] = [];
+                for (const ol of otpOrLink) {
+                    for (const ep of emailOrPhone) {
+                        allFactors.push(`${ol}-${ep}`);
+                    }
+                }
+
+                PostSuperTokensInitCallbacks.addPostInitCallback(() => {
+                    const mfaInstance = MultiFactorAuthRecipe.getInstance();
+
+                    if (mfaInstance !== undefined) {
+                        mfaInstance.addGetAllFactorsFromOtherRecipesFunc((tenantConfig) => {
+                            if (tenantConfig.passwordless.enabled === false) {
+                                return [];
+                            }
+                            return allFactors;
+                        });
+                        mfaInstance.addGetFactorsSetupForUserFromOtherRecipes(async (_: string, user: User) => {
+                            // We deliberately do not check for matching tenantId because
```

Comment:
i think we need to check for tenantId here too, cuase if we don't things don't work in the following case:
- tenant1 -> passwordless user with email e1
- tenant2 -> emailpassword user with email e1

User logs into tenant2 with email password and needs to do otp-email. Now we return that passwordless is already setup, and this way, the consumeCode API is called with e1. Now the API, being tenant specific, will try to do a sign up, and check that factor setup is allowed -> this will return false now!

If we make this tenant specific though, then the same thing will happen (from the user's point of view), except that the consumeCode API, when it checks that is factorSetup allowed, then it will be allowed cause it's not setup!

Reply:
I wonder why this did not come up in testing..

Reply:
Same thing goes for other recipes.